src/Gui/PreferencePackTemplates/View.cfg

```diff
@@ -14,6 +14,9 @@
         </FCParamGroup>
         <FCParamGroup Name="View">
           <FCInt Name="AntiAliasing" Value="0"/>
+          <FCUInt Name="AxisXColor" Value="3425907456"/>
+          <FCUInt Name="AxisYColor" Value="869020416"/>
+          <FCUInt Name="AxisZColor" Value="859032576"/>
```

Comment:
I have a hard time figuring out how to set default preferences. Here? For some reason, the X color is sometimes unset upon preference reset. Anyone here to help out? @qewer33?

Reply:
I'm not sure how the preference reset system works. Maybe @CalligaroV, the person who implemented part of the preference settings reset buttons can help?

Reply:
If I'm not mistaken this is used as a template for creating your own PreferencePack, not used for resetting, I believe when you use the reset button it removes the preference entry rather than setting it to a default value.

Reply:
If I'm not mistaken this is used as a template for creating your own PreferencePack, not used for resetting, I believe when you use the reset button it removes the preference entry rather than setting it to a default value.

Reply:
Hi @qewer33 and @benj5378!

AFAICS, as those Parameters/Preferences use `Gui::PrefColorButton`, it should be enough to add something like

```c++
void DlgSettings3DViewImp::resetSettingsToDefaults()
{
    //add here other parameters that don't use Gui::Pref* widgets

    //finally reset all the parameters associated to Gui::Pref* widgets
    PreferencePage::resetSettingsToDefaults();
}
```
in `DlgSettings3DViewImp.cpp` and the logic should take care of handling the reset request, resetting the value to what's specified in the `.ui` file.

For reference, you can see that something similar is made elsewhere for similar parameters, for example in https://github.com/FreeCAD/FreeCAD/blob/32e09d9554db26414697ed19e0189ccab4ea9029/src/Gui/PreferencePages/DlgSettingsNavigation.cpp#L236

In that case there are the parameters `BaseColor` and `RotationCenterColor` that should be handled by the reset logic (don't remember if I tested them)

I don't know if in this specific case there may be something that doesn't make the reset logic work correctly but in other places it's working (or better, I'm not aware of issues about wrong/missed resets)